Sources/Rendering/Core/CubeAxesActor/index.js

```diff
@@ -0,0 +1,830 @@
+        publicAPI.createPolyDataForOneLabel(
+          model.textValues[textIdx],
+          ptIdx,
+          cmat,
+          invmat,
+          dir,
+          model.tickLabelPixelOffset,
+          results
+        );
+        ptIdx++;
+        textIdx++;
+      }
+      axisIdx++;
+    }
+
+    const tcoordDA = vtkDataArray.newInstance({
+      numberOfComponents: 2,
+      values: tcoords,
+      name: 'TextureCoordinates',
+    });
+    model.tmPolyData.getPointData().setTCoords(tcoordDA);
+    model.tmPolyData.getPoints().setData(points, 3);
+    model.tmPolyData.getPoints().modified();
+    model.tmPolyData.getPolys().setData(polys, 1);
+    model.tmPolyData.getPolys().modified();
+    model.tmPolyData.modified();
+  };
+
+  publicAPI.getActors = () => [model.actor, model.pixelActor, model.tmActor];
+
+  publicAPI.getNestedProps = () => publicAPI.getActors();
+
+  publicAPI.setGridLines = (val) => {
+    if (model.gridLines === val) {
+      return;
+    }
+    model.gridLines = val;
+    publicAPI.update(true);
+    publicAPI.modified();
+  };
+
+  publicAPI.setAxisLabels = (...args) => {
+    let array = args;
+    // allow an array passed as a single arg.
+    if (array.length === 1 && Array.isArray(array[0])) {
+      /* eslint-disable prefer-destructuring */
+      array = array[0];
+      /* eslint-enable prefer-destructuring */
+    }
+
+    if (array.length !== 3) {
+      throw new RangeError(
+        `Invalid number of values for array setter (AxisLabels)`
+      );
+    }
+
+    const changeDetected = model.axisLabels.some(
+      (item, index) => item !== array[index]
+    );
+
+    if (changeDetected || model.axisLabels.length !== array.length) {
+      model.axisLabels = [].concat(array);
+      publicAPI.update(true);
+      console.log('forcing update');
+      publicAPI.modified();
+      return true;
+    }
+    return false;
+  };
+
+  publicAPI.setTickTextStyle = (tickStyle) => {
+    model.tickStyle = { ...model.tickStyle, ...tickStyle };
+    publicAPI.update(true);
+    publicAPI.modified();
+  };
+
+  publicAPI.setAxisTextStyle = (axisStyle) => {
+    model.axisStyle = { ...model.axisStyle, ...axisStyle };
+    publicAPI.update(true);
+    publicAPI.modified();
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+export const DEFAULT_VALUES = {
+  camera: null,
+  dataBounds: null,
+  gridLines: true,
+  axisLabels: null,
+  axisTitlePixelOffset: 35.0,
+  axisTextStyle: {
+    fontColor: 'white',
+    fontStyle: 'normal',
+    fontSize: 18,
+    fontFamily: 'serif',
+    strokeColor: 'black',
+    strokeSize: 1,
+    lineSpace: 0.2,
+  },
+  tickLabelPixelOffset: 12.0,
+  tickTextStyle: {
+    fontColor: 'white',
+    fontStyle: 'normal',
+    fontSize: 14,
+    fontFamily: 'serif',
+    strokeColor: 'black',
+    strokeSize: 1,
+    lineSpace: 0.2,
+  },
+};
```

Comment:
You can't have objects in `DEFAULT_VALUES`, see https://github.com/Kitware/vtk-js/issues/1751#issuecomment-785411007 for an alternative to `DEFAULT_VALUES`

Reply:
Ha, I've been bitten by that before.  I lifted that code from label representation so that probably needs to be fixed.